dspace-api/src/main/java/org/dspace/fileaccess/FileAccessConsumer.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.fileaccess;
+
+import java.util.*;
+import org.apache.log4j.*;
+import org.dspace.content.*;
+import org.dspace.content.factory.*;
+import org.dspace.content.service.*;
+import org.dspace.core.*;
+import org.dspace.event.*;
+import org.dspace.fileaccess.factory.*;
+import org.dspace.fileaccess.service.*;
+
+/**
+ * Created by: Antoine Snyers (antoine at atmire dot com)
+ * Date: 02 Oct 2015
+ */
+public class FileAccessConsumer implements Consumer {
+
+    /**
+     * log4j logger
+     */
+    private static Logger log = Logger.getLogger(FileAccessConsumer.class);
+
+    private List<Item> items = new LinkedList<>();
+
+    protected ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+    protected BitstreamService bitstreamService = ContentServiceFactory.getInstance().getBitstreamService();
+    protected FileAccessFromMetadataService fileAccessFromMetadataService = FileAccessServiceFactory.getInstance().getFileAccessFromMetadataService();
+
+    public void initialize() throws Exception {
+
+    }
+
+    /**
+     * Gather the DspaceObject IDs here.
+     * DO NOT COMMIT THE CONTEXT
+     */
+    public void consume(Context context, Event event) throws Exception {
+
+
+        int subjectType = event.getSubjectType();
+        int eventType = event.getEventType();
+
+        switch (subjectType) {
+            case Constants.ITEM:
+                if (eventType == Event.INSTALL) {
+                    items.add((Item) event.getSubject(context));
+                }
+                break;
+            default:
+                log.warn("consume() got unrecognized event: " + event.toString());
```

Comment:
I'm afraid this could spam the logs. I made bad experiences with logging when developing the DOIConsumer. Please reduce the log level to debug.